core/tests/org.fusesource.ide.camel.model.service.core.tests/src-test/org/fusesource/ide/camel/model/service/core/util/PropertiesUtilsTest.java

```diff
@@ -0,0 +1,83 @@
+package org.fusesource.ide.camel.model.service.core.util;
+
+import static org.fusesource.ide.camel.model.service.core.util.PropertiesUtils.replaceParts;
+import static org.fusesource.ide.camel.model.service.core.util.PropertiesUtils.updatePathParams;
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.fusesource.ide.camel.model.service.core.catalog.Parameter;
+import org.junit.Test;
+
+/**
+ * 
+ * @author apodhrad
+ *
+ */
+public class PropertiesUtilsTest {
+
+	@Test
+	public void testUpdatePathParams() {
+		String syntax = "timer:timerName";
+		List<Parameter> pathParams = pathParams("timerName");
+		Map<String, String> modelMap = new HashMap<String, String>();
+		modelMap.put("timerName", "timerName");
+
+		String expected = "timer:coolName";
+		assertEquals(expected, updatePathParams(syntax, pathParam("timerName"), "coolName", pathParams, modelMap));
+
+		syntax = "sap-idoclist-destination:destination:idocType:idocTypeExtension:systemRelease:applicationRelease";
+		pathParams = pathParams("destination", "idocType", "idocTypeExtension", "applicationRelease");
+		modelMap = new HashMap<String, String>();
+		modelMap.put("destination", "destination");
+		modelMap.put("idocType", "idocType");
+		modelMap.put("idocTypeExtension", "idocTypeExtension");
+		modelMap.put("applicationRelease", "applicationRelease");
+
+		expected = "sap-idoclist-destination:myDest:idocType:idocTypeExtension:systemRelease:applicationRelease";
+		assertEquals(expected, updatePathParams(syntax, pathParam("destination"), "myDest", pathParams, modelMap));
+
+		expected = "sap-idoclist-destination:destination:abc:idocTypeExtension:systemRelease:applicationRelease";
```

Comment:
you should better write separated unit tests for each conditions.
It helps to have a concise view of what is failing exactly
You can @Before method for the initialization or just extract a method
It also helps to understand each tested case as we need to provide a new test method name for each case :-)